hr_expense_petty_cash/models/account_invoice.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2019 Ecosoft Co., Ltd. (http://ecosoft.co.th)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models, fields, _
+from odoo.exceptions import ValidationError
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    is_petty_cash = fields.Boolean(
+        string='Petty Cash',
+        default=False,
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+    )
+
+    @api.multi
+    @api.constrains('invoice_line_ids')
+    def _check_petty_cash_amount(self):
+        for rec in self:
+            if not rec.is_petty_cash:
+                continue
+            petty_cash = self.env['petty.cash'].search(
+                [('partner_id', '=', self.partner_id.id)], limit=1)
+            if not petty_cash:
+                raise ValidationError(_('%s is not a petty cash holder') %
+                                      self.partner_id.name)
+            balance = petty_cash.petty_cash_balance
+            limit = petty_cash.petty_cash_limit
+            max_amount = limit - balance
+            account = petty_cash.account_id
+            # Looking for petty cash line
+            inv_lines = rec.invoice_line_ids.filtered(
+                lambda l: l.account_id == account)
+            amount = sum([line.price_subtotal for line in inv_lines])
+            company_currency = rec.company_id.currency_id
+            amount_currency = rec.currency_id.compute(amount, company_currency)
```

Comment:
```suggestion
            amount_company = rec.currency_id.compute(amount, company_currency)
```
I understand that you try to change from any currency (USD), into company currency (THB), right?
If so, amount in company currency, should named, amount_company. This is naming convention you can find in account.move.line, i.e.,
* debit, credit = company currency always, i.e., THB (else where it will be amount_company)
* amount_currency = foreign currency, i.e., USD